File: csharp/difference-of-squares/DifferenceOfSquares.cs
using System;
using System.Linq;

public class Squares
{
  private int squares;
  private int[] range;

  public Squares(int value)
  {
    if (value < 0) 
    {  
      throw new ArgumentException("Input must be greater than or equal to 0.");
    }
    
    squares = value;
    range = Enumerable.Range(1, value).ToArray();
  }

  public int SquareOfSums()
  {
    return squares == 0 ? 0 : (int)Math.Pow(range.Aggregate((a, b) => a + b), 2);
  }

  public int SumOfSquares()
  {
    return squares == 0 ? 0 : (int)range.Select(x => Math.Pow(x, 2)).Sum();
  }

  public int DifferenceOfSquares()
  {
    return SquareOfSums() - SumOfSquares();
  }
}

File: javascript/difference-of-squares/difference-of-squares.js
export class Squares {
  constructor(n) {
    this._squares = n;
  }

  get squareOfSum() {
    if (this._squares > 0) {
      let sum = 0;
      for (let i = 1; i <= this._squares; i++) {
        sum += i;
      }
      return Math.pow(sum, 2);
    }
    return 0;
  }

  get sumOfSquares() {
    if (this._squares > 0) {
      let sum = 0;
      for (let i = 1; i <= this._squares; i++) {
        sum += Math.pow(i, 2);
      }
      return sum;
    }
    return 0;
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}
